src/main/java/org/christiangalsterer/stash/filehooks/plugin/hook/FileNameHook.java

```diff
@@ -122,4 +145,82 @@ public void validate(Settings settings, SettingsValidationErrors errors, Reposit
             }
         }
     }
+	
+	  /**
+     * Callback, collecting all the paths, changed in the requested change 
+     * range. 
+     */ 
+    private static class ChangesPathsCollector extends AbstractChangeCallback { 
+        private final Collection<String> changedPaths = new HashSet<String>(); 
+ 
+        @Override 
+        public boolean onChange(Change change) throws IOException {
+        	//isNotDeleteChange
+        	if ( change.getType() != ChangeType.DELETE )
+        	{
+        		changedPaths.add(change.getPath().toString());
+        	}
+            return true; 
+        } 
+ 
+        public Collection<String> getChangedPaths() { 
+            return changedPaths; 
+        } 
+ 
+    }  
+  			 	
+	 private String getPullRequestError(Collection<String> filteredFiles) { 
+	        final StringBuilder sb = new StringBuilder(); 
+	        sb.append(i18n.getText("filename-hook.error.pattern", "Pattern is not a valid regular expression")); 
+	        sb.append("\n"); 
+	        final Iterator<String> iter = filteredFiles.iterator(); 
+	        while (iter.hasNext()) { 
+	            sb.append(iter.next()); 
+	            if (iter.hasNext()) { 
+	                sb.append(", "); 
+	            } 
+	        } 
+	        return sb.toString(); 
+	    } 
+	 
+	@Override
+	public void check(RepositoryMergeRequestCheckContext context) {
+		final MergeRequest request = context.getMergeRequest();
+		final PullRequest pr = request.getPullRequest();
+		final Commit prFrom = getChangeSet(pr.getFromRef());
+		final Commit prTo = getChangeSet(pr.getToRef());
+		final Commit base = mergeBaseResolver.findMergeBase(prFrom, prTo);
+	    final FileNameHookSetting setting = getSettings(context.getSettings());
+	    
+		final ChangesRequest.Builder builder = new ChangesRequest.Builder(prFrom.getRepository(), prFrom.getId()); 
+        if (base.getId() != null) { 
+            builder.sinceId(base.getId()); 
+        } 
+        final ChangesRequest pathsRequest = builder.build(); 
+        final ChangesPathsCollector pathsCallback = new ChangesPathsCollector(); 
+        commitService.streamChanges(pathsRequest, pathsCallback); 
+	    
+	    
+	    Collection<String> filteredFiles = pathsCallback.getChangedPaths(); 
+		 
+		
+	    filteredFiles = Collections2.filter(filteredFiles, Predicates.contains(setting.getIncludePattern()));	
+		 
+		if(setting.getExcludePattern().isPresent()) {
+			 Pattern excludePattern = setting.getExcludePattern().get();
+			 filteredFiles = Collections2.filter(filteredFiles, Predicates.not(Predicates.contains(excludePattern)));
+		}
+	    
+	    if (filteredFiles.size() > 0) {
```

Comment:
Does "filename-hook.error, "File violates file name pattern for branch"  is fine?